src/views/virtualmachines/actions/components/CloneVMModal/CloneVMModal.tsx

```diff
@@ -41,90 +26,67 @@ type CloneVMModalProps = {
 
 const CloneVMModal: FC<CloneVMModalProps> = ({ isOpen, onClose, vm }) => {
   const { t } = useKubevirtTranslation();
-
   const history = useHistory();
 
-  const [cloneName, setCloneName] = useState(`${vm?.metadata?.name}-clone`);
-  const [cloneDescription, setCloneDescription] = useState(
-    vm?.metadata?.annotations?.[DESCRIPTION_ANNOTATION],
+  const [cloneName, setCloneName] = useState(
+    `${vm?.metadata?.name}-clone-${getRandomChars()}`.substring(0, MAX_K8S_NAME_LENGTH),
   );
-  const [cloneProject, setCloneProject] = useState(vm?.metadata?.namespace);
-  const [startCloneVM, setStartCloneVM] = useState(false);
 
-  const isVMRunning = isRunning(vm);
+  const [vmsInSameNamespace] = useK8sWatchResource<V1VirtualMachine[]>(
+    vm && {
+      groupVersionKind: modelToGroupVersionKind(VirtualMachineModel),
+      isList: true,
+      namespace: vm.metadata.namespace,
+    },
+  );
+
+  const nameAlreadyUsed = Boolean(vmsInSameNamespace?.find((v) => v.metadata.name === cloneName));
```

Comment:
2 things.
1. why bring them all then filter? try to fetch with the name and namespace if result is 200 its existing. 
2. u really think u need it? what are the chances that machine will be existing? and if existing do u display the error from BE? u will get 409 conflict that you can present.

Reply:
1. Nice suggestion we can do that even if this means that there will be a loading after typing.
2 There is no error from BE if you use the name of an existing VM. If it exists, the BE will restore the existing VM using the selected VM state and this is something we should absolutely avoid